src/main/java/com/meetup/server/event/dto/response/RouteResponse.java

```diff
@@ -0,0 +1,60 @@
+package com.meetup.server.event.dto.response;
+
+import com.meetup.server.global.clients.kakao.mobility.KakaoMobilityResponse;
+import com.meetup.server.global.clients.odsay.OdsayTransitRouteSearchResponse;
+import com.meetup.server.startpoint.domain.StartPoint;
+import com.meetup.server.user.domain.User;
+import lombok.Builder;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Builder
+public record RouteResponse(
+        boolean isTransit,  // true: 대중교통, false: 자동차
+        UUID id,
+        String nickname,
+        String profileImage,
+        String startName,  //출발지 주소
+        double startLongitude,  //실제 출발지 경도(:longitude)
+        double startLatitude,  //실제 출발지 위도(:latitude)
+        List<TransitRouteResponse> transitRoute,
+        List<DrivingRouteResponse> drivingRoute,
+        int totalTime
+
+) {
+    public static RouteResponse of(StartPoint startPoint,
+                                   User user,
+                                   OdsayTransitRouteSearchResponse transitResponse,
+                                   KakaoMobilityResponse drivingResponse,
+                                   boolean isTransit) {
+        return RouteResponse.builder()
+                .isTransit(isTransit)
+                .id(startPoint.getStartPointId())
+                .nickname(startPoint.getIsUser() ? user.getNickname() : startPoint.getNonUserName())
+                .profileImage(startPoint.getIsUser() ? user.getProfileImage() : null)
+                .startName(convertStartPointName(startPoint.getAddress().getAddress()))
+                .startLongitude(startPoint.getLocation().getRoadLongitude())
+                .startLatitude(startPoint.getLocation().getRoadLatitude())
+                .transitRoute(TransitRouteResponse.from(transitResponse))
+                .drivingRoute(DrivingRouteResponse.from(drivingResponse))
+                .totalTime(isTransit ?
+                        transitResponse.data().path().getFirst().info().totalTime()
+                        : drivingResponse.routes().getFirst().summary().duration())
+                .build();
+    }
+
+    private static String convertStartPointName(String address) {
+        if (address == null || address.isBlank()) return "";
+
+        Pattern pattern = Pattern.compile("(\\S+(구|군))\\s+(\\S+(동|읍))");
```

Comment:
`Pattern.compile()`을 하게 되면 매번 Pattern 인스턴스가 생성되고 바로 버려져요!
상수로 표현해서 인스턴스가 한번만 생성되도록 변경하면 좋을 거 같아요! ([아이템 6 참고](https://developer-anxi.tistory.com/41))

```suggestion
        private static final Pattern pattern = Pattern.compile("(\\S+(구|군))\\s+(\\S+(동|읍))");
```

Reply:
추가로 프론트에서 처리할 수도 있는 부분인 만큼, 원본주소를 백에서 넘기고 프론트에서 가공하는게 좋을지도 논의해보면 좋을 거 같습니다~